pkg/multicluster/kubeconfig/loader.go

```diff
@@ -11,39 +11,50 @@ import (
+// Override the API Server URL and current context if specified.
+// Copied and modified from https://github.com/kubernetes-sigs/controller-runtime/blob/cb7f85860a8cde7259b35bb84af1fdcb02c098f2/pkg/client/config/config.go#L135
+func GetClientConfigWithContext(kubeConfigPath, kubeContext, apiServerUrl string) (clientcmd.ClientConfig, error) {
 	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
-	if _, ok := os.LookupEnv("HOME"); !ok {
-		u, err := user.Current()
-		if err != nil {
-			return nil, fmt.Errorf("could not get current user: %v", err)
+
+	if kubeConfigPath != "" {
+		loadingRules.ExplicitPath = kubeConfigPath
+	} else {
+		// Fetch kubeconfig from environment in which this is invoked
+		if _, ok := os.LookupEnv("HOME"); !ok {
+			u, err := user.Current()
+			if err != nil {
+				return nil, fmt.Errorf("could not get current user: %v", err)
+			}
+			loadingRules.Precedence = append(loadingRules.Precedence, path.Join(u.HomeDir, clientcmd.RecommendedHomeDir, clientcmd.RecommendedFileName))
+		}
+	}
+
+	overrides := &clientcmd.ConfigOverrides{}
+	if kubeContext != "" {
+		overrides.CurrentContext = kubeContext
+	}
+	if apiServerUrl != "" {
+		overrides.ClusterInfo = clientcmdapi.Cluster{
+			Server: apiServerUrl,
 		}
-		loadingRules.Precedence = append(loadingRules.Precedence, path.Join(u.HomeDir, clientcmd.RecommendedHomeDir, clientcmd.RecommendedFileName))
 	}
-	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
-		loadingRules,
-		&clientcmd.ConfigOverrides{
-			ClusterInfo: clientcmdapi.Cluster{
-				Server: apiServerUrl,
-			},
-			CurrentContext: kubeContext,
-		}), nil
+
+	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, overrides), nil
 }
 
-// Fetch rest.Config for environment in which this is invoked, override the API Server URL and current context.
-func GetRestConfigWithContext(apiServerUrl, kubeContext string) (*rest.Config, error) {
-	clientConfig, err := GetClientConfigWithContext(apiServerUrl, kubeContext)
+// Fetch rest.Config for environment in which this is invoked, override the API Server URL and current context if specified.
+func GetRestConfigWithContext(kubeConfigPath, kubeContext, apiServerUrl string) (*rest.Config, error) {
+	clientConfig, err := GetClientConfigWithContext(kubeConfigPath, kubeContext, apiServerUrl)
 	if err != nil {
 		return nil, err
 	}
 	return clientConfig.ClientConfig()
 }
 
-// Fetch raw Config for environment in which this is invoked, override the API Server URL and current context.
-func GetRawConfigWithContext(apiServerUrl, kubeContext string) (clientcmdapi.Config, error) {
-	clientConfig, err := GetClientConfigWithContext(apiServerUrl, kubeContext)
+// Fetch raw Config for environment in which this is invoked, override the API Server URL and current context if specified.
+func GetRawConfigWithContext(kubeConfigPath, kubeContext, apiServerUrl string) (clientcmdapi.Config, error) {
```

Comment:
WDYT about deprecating or getting rid of these helpers in favor of `GetClientConfigWithContext`? On the one hand it's nice to have a function signature we can quickly reference to find the type of kubeconfig we need, but on the other these create an additional maintenance burden on top of GetClientConfigWithContext.

Reply:
I could go either way. I lean slightly towards keeping this convenience helpers since I don't expect we'll ever have to update them

Reply:
works for me! I think having these will help folks who are less familiar with the various kube config types find what they need a bit quicker.